framework/src/problems/FEProblemBase.C

```diff
@@ -2517,10 +2525,37 @@ FEProblemBase::addKernel(const std::string & kernel_name,
     parameters.set<SystemBase *>("_sys") = _nl[nl_sys_num].get();
   }
 
-  logAdd("Kernel", name, kernel_name);
+  logAdd(base_name, name, ro_name);
+}
+
+void
+FEProblemBase::addKernel(const std::string & kernel_name,
+                         const std::string & name,
+                         InputParameters & parameters)
+{
+  parallel_object_only();
+  const auto nl_sys_num =
+      determineNonlinearSystem(parameters.varName("variable", name), true).second;
+  setResidualObjectParamsAndLog(
+      kernel_name, name, parameters, nl_sys_num, "Kernel", _reinit_displaced_elem);
+
   _nl[nl_sys_num]->addKernel(kernel_name, name, parameters);
 }
 
+void
+FEProblemBase::addHDGKernel(const std::string & kernel_name,
+                            const std::string & name,
+                            InputParameters & parameters)
+{
+  parallel_object_only();
+  const auto nl_sys_num =
+      determineNonlinearSystem(parameters.varName("variable", name), true).second;
+  setResidualObjectParamsAndLog(
+      kernel_name, name, parameters, nl_sys_num, "HDGKernel", _reinit_displaced_elem);
+
```

Comment:
same

Reply:
that's done in `setResidualObjectParamsAndLog`